### ISSUE TITLE
Symfony PHP Services Config

### ISSUE THREAD

**Issue by natepage** [eonx-com/easy-monorepo#268]
In preparation of Symfony 6, an initiative made by the core team was to move XML and YAML services config to PHP (https://github.com/symfony/symfony/issues/37186)

Let's do the same and get ready for the furure! 😄 

This will require us to:

- Change the DI file loader to use PHPFileLoader
- Replace all `Bridge/Symfony/Resources/config/*.yaml` and `*.xml` files to `*.php`

Here is an example: https://github.com/symfony/symfony/commit/18fcb5f956779cf6faaafe5d2087c80005182202

**Comment by BOB41K1987** [comment]
Here is a list of config files that need to be changed:

- [ ] packages/EasyApiToken/src/Bridge/Symfony/Resources/config/services.xml
- [ ] packages/EasyAsync/src/Bridge/Symfony/Resources/config/implementations/doctrine.yaml
- [ ] packages/EasyAsync/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasyAwsCredentialsFinder/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/debug.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/iri_converter.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/maker_commands.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/no_properties_api_resource.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/pagination.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/api_platform/simple_data_persister.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/easy_async_listeners.yaml
- [ ] packages/EasyCore/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasyDecision/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasyDecision/src/Bridge/Symfony/Resources/config/use_expression_language.yaml
- [ ] packages/EasyPagination/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasyPsr7Factory/src/Bridge/Symfony/Resources/config/services.xml
- [ ] packages/EasyRandom/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasySchedule/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasySecurity/src/Bridge/Symfony/Resources/config/services.yaml
- [ ] packages/EasySsm/config/console_loader.yaml
- [ ] packages/EasySsm/config/dotenv_loader.yaml
- [ ] packages/EasySsm/config/services.yaml

**Comment by BOB41K1987** [comment]
@natepage those new changes require new PHP DSL functions that are going to be available in symfony 5.2, symfony/dependency-injection specifically. Let's wait for symfony 5.2 release.

**Comment by natepage** [comment]
@BOB41K1987 I've used this new syntax in the new `eonx-com/easy-event-dispatcher` I've created yesterday and it is working with Symfony 4.4, which functions are you talking about?

**Comment by TomasVotruba** [comment]
Just a tip how to automate it: https://tomasvotruba.com/blog/2020/07/16/10-cool-features-you-get-after-switching-from-yaml-to-php-configs/

**Comment by BOB41K1987** [comment]
@natepage I'm talking about `service`, `param`, `abstract_arg` and other functions that are not available yet. They seems to be added pretty recently https://github.com/symfony/symfony/pull/36775 and not yet available.

@TomasVotruba thanks for the tip, we will definitely check that out!

**Comment by TomasVotruba** [comment]
@BOB41K1987 

> I'm talking about service, param, abstract_arg and other functions that are not available yet. They seems to be added pretty recently symfony/symfony#36775 and not yet available.

Good catch! [The `service()` was `ref()` until Symfony 5.1](https://github.com/symfony/symfony/pull/36800). To migrate to specific Symfony version, there i `--target-symfony-version` in the tool.

**Comment by natepage** [comment]
@TomasVotruba Thanks for the tip and tool! 😄 

@BOB41K1987 Not all our projects are running on the latest Symfony version, I think we should still proceed with refactoring using the DSL from 4.4 and we will use the new functions when we drop the support for 4.4

**Comment by natepage** [comment]
@BOB41K1987 I've ticked the xml file from EasyApiToken as I've done it in #286 

Any progress on the other files?

**Comment by TomasVotruba** [comment]
Why not all at one run? 

I just posted "tutorial" yesterday: https://tomasvotruba.com/blog/2020/07/27/how-to-switch-from-yaml-xml-configs-to-php-today-with-migrify/

**Comment by natepage** [comment]
@TomasVotruba I've read that post 😄 And all config at single run sounds good, just trying not to be the only one working on that repo haha

@BOB41K1987 Will take care of that like a champ I'm sure

**Comment by TomasVotruba** [comment]
I see. I wish I had to little work too :D I have quite contraty problem

**Comment by natepage** [comment]
@TomasVotruba you're too famous, we don't have that problem here (yet 😛), hope Rector is going ok

**Comment by TomasVotruba** [comment]
In that case I should tell you I'm looking 1-2 developers for Rector :D

**Comment by natepage** [comment]
I wish we could help but atm we're also looking for new devs. Once you're done with Rector refactoring legacy code, you should probably work on Creator to create the legacy code of tomorrow 😄 This way we don't need devs for the actual projects anymore 😄 

I'm currently listening to "The Inevitable" and it looks like we're going this way already.

What's next on Rector's roadmap? I unfortunately don't have resources available to work full time on it, but if it can help our business on the long run somehow, it wouldn't hurt to dedicate some time to contribute to open-source projects like Rector

**Comment by TomasVotruba** [comment]
>  This way we don't need devs for the actual projects anymore smile

That's exactly what I had in mind last few days. Many companies waste resources on wrong places creating artificial work that cost them money and time. Check discussion here: https://twitter.com/VotrubaT/status/1287349364676067328

> I'm currently listening to "The Inevitable" and it looks like we're going this way already.

Great book! I've followed it with "Give & Take", on more human note :)

> What's next on Rector's roadmap?

Symfony PHP configs opens completely new field to refactor. We already added & used 1st rule to change `arg()` to `call()`: https://github.com/rectorphp/rector/blob/master/docs/rector_rules_overview.md#changeserviceargumentstomethodcallrector

Last day I spend on refatoring stringy configuration :vomiting_face: to constants and `configure()` method contract: https://github.com/rectorphp/rector/pull/3821 Rector was pretty shit in configuration before :D

**Comment by natepage** [comment]
> Many companies waste resources on wrong places creating artificial work that cost them money and time.

Couldn't agree more, I've worked on a lot of different projects, small and big, and each time it is the same feeling... Different teams, solving the same problems in different ways 😄 

That's exactly why this monorepo came to life, create reusable packages for all the "boring" stuff and be able to focus on the business logic, the real value!

We had a few attempts of creating a "skeleton" we used as starting point for new projects but it was really hard to keep it up-to-date at the same time of the actual projects. 

Might be where Rector comes into the picture!
